cloud/ec2_util.go

```diff
@@ -791,6 +791,14 @@ func canUseElasticIP(settings *evergreen.Settings, ec2Settings *EC2ProviderSetti
 // allocateIPAddressForHost allocates an elastic IP address from an IPAM pool
 // for the host.
 func allocateIPAddressForHost(ctx context.Context, settings *evergreen.Settings, c AWSClient, h *host.Host) error {
+	flags, err := evergreen.GetServiceFlags(ctx)
+	if err != nil {
+		return errors.Wrap(err, "getting service flags")
+	}
+	if flags.ElasticIPsDisabled {
+		return nil
```

Comment:
Here we return nil which would be a no-op (but the other one we return an error for example)